runtime/kusama/Cargo.toml

```diff
@@ -70,7 +70,7 @@ pallet-recovery = { git = "https://github.com/paritytech/substrate", branch = "m
 pallet-referenda = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
 pallet-scheduler = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
 pallet-session = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
-pallet-society = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false }
+pallet-society = { git = "https://github.com/paritytech/substrate", branch = "master", default-features = false, features = ["experimental"] }
```

Comment:
Not sure we should enable them right now. These features will bleed everywhere.

Reply:
Maybe we should probably have a discussion if we want to use `experimental` in our prod runtimes? 

My understanding is we added `experimental` in large part to give us licence to iterate on APIs we anticipate will change but still want to use internally without it causing issues with our LTS versioning, so they should be fine to use here. But open to discussion :)

Reply:
There is not LTS versioning coming. The LTS is about backporting security fixes to this LTS release and not more. We will not backport anything else.

Reply:
> Maybe we should have a discussion if we want to use `experimental` in our prod runtimes?

I mean the experimental feature makes sense for stuff that is experimental and shouldn't be used yet. Especially for stuff in macros that we not gate in any other way. However, for these things which are working and may just change api, we could put into some "experimental" module.

Reply:
I think the spirit of creating the experimental feature was to introduce stuff that works but the API may break, not something that is experimental and shouldn't be used yet. It was decided to use a feature flag instead of a module here: https://github.com/paritytech/substrate/issues/14345 and my understanding was that VersionedRuntimeUpgrade would be used in the prod Polkadot runtimes even while experimental, and we would shift it out of experimental once the API is stabalised, but I'll let @ggwpez and @kianenigma chime in here to confirm/deny this assumption.

My 2c on code that is so experimental that it should not be used is that 99% of the time it probably shouldn't be on the main branch but rather live a feature branch until it is ready to be used?

Reply:
> However, for these things which are working and may just change api, we could put into some "experimental" module.

That was the idea with the feature flag, but now seeing how it bleeds through is annoying. But wouldnt it only affect crates that depend on the `kusama-runtime` in this case? Or if someone uses the `--workspace` flag i assume.

So you mean we should have experimental flag and module?

Reply:
Can either of you explain in detail this "feature bleeding" phenomenon and exactly how it will negatively impact this process? 

From the name, I infer that in any higher level crate using `pallet-society` (eg.  `kusama-runtime`), even if the higher level crate imports `frame-support` normally, because `pallet-society` has `frame-support/experimental`, the latter will overrule. 

This seems fine to me for now. As long as this feature is inly adding types and not tweaking behaviors in obscure ways, I cannot see it going wrong. 

my 2c against a mod is that it would make the process of moving from `experimental` to `stable` a breaking change as the import path now changes.

Reply:
> From the name, I infer that in any higher level crate using `pallet-society` (eg. `kusama-runtime`), even if the higher level crate imports `frame-support` normally, because `pallet-society` has `frame-support/experimental`, the latter will overrule.

Exactly.

> This seems fine to me for now. As long as this feature is inly adding types and not tweaking behaviors in obscure ways, I cannot see it going wrong.

Because the stuff is behind the `experimental` flag means that the signature or whatever of it can change at any point. Builders that depend on Polkadot will currently pull in at least the `polkadot-runtime` crate (There is a pr open to not pull in the crate) and then have the feature enabled. The will then be able to use the feature, without maybe realizing that it is tagged as experimental. So the only downside is maybe that their code breaks at some update.

Reply:
> The will then be able to use the feature, without maybe realizing that it is tagged as experimental.

Hmm this is a good point. I will update the doc comment to indicate it is experimental, hopefully reducing the chance of this happening.